### DISCUSSION TITLE
fullyQualifiedName may return wrong import

### DISCUSSION THREAD

**Discussion by Rawa** [LemonAppDev/konsist#872] [Report a Problem] [answered]
### Tell us what version of Konsist you use.

0.13.0

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

fullyQualifiedName sometimes returns wrong import. I was writing a test to make some assertions on all my Test functions. Thus I wanted to grab all the functions that were annotated with `@Test`, the helper functions of grabbing by annotation did not work, thus I inspected one of the function and was able to see that the properties `name`, `text` and `location` was correct for the annotation but the `fullyQualifiedName` was a completely other class, in my case `kotlinx.coroutines.test.UnconfinedTestDispatcher`

https://kotlinlang.slack.com/archives/C05QG9FD6KS/p1708527727842249

### Actual behaviour

`fullyQualifiedName` returns the first import matching the annotation name.

### Expected behaviour

`fullyQualifiedName` should return the exact match of import

**Comment by Rawa** [comment]
I believe the issue is caused by this line:
https://github.com/LemonAppDev/konsist/blob/12eb98c15a41be02b100e02e4c3cc74db213f35c/lib/src/main/kotlin/com/lemonappdev/konsist/core/provider/KoFullyQualifiedNameProviderCore.kt#L19C1-L19C60

We simply grabs the first import that contains the annotations name.

**Reply by Rawa** [reply]
Maybe a better implementation would be something along the lines of:

```
    override val fullyQualifiedName: String
        get() {
            var fqn = containingFile
                .imports
                .map {
                    it.name.split(".").last()
                }
                .firstOrNull { it == textUsedToFqn }
            }
            ...
```

However, I'm not sure if we have any strategy for handling the Wildcard annotation? Is there a way to expand and find all the import and match based on that?

**Comment by Rawa** [comment]
Found now that it has been resolved by this PR
https://github.com/LemonAppDev/konsist/discussions/738 I'll close this discussion.

**Comment by nataliapeterwas** [comment, marked as answer]
A new version v0.14.0 has been released. It contains the fix.
See [v0.14.0 release](https://github.com/LemonAppDev/konsist/releases/tag/v0.14.0)
